onnxruntime/core/providers/cuda/cuda_utils.cu

```diff
@@ -10,6 +10,9 @@
 namespace onnxruntime {
 namespace cuda {
 
+std::once_flag GridDim::s_cachedDevicePropsInitFlag;
```

Comment:
Add some comments for s_cachedDevicePropsInitFlag and s_cachedDeviceProps. Why do they need to be global variable?

Reply:
> Add some comments for s_cachedDevicePropsInitFlag and s_cachedDeviceProps. Why do they need to be global variable?

Those 2 static variable of GridDim is declared in class GridDim, but not defined.